src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,17 +27,24 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
+        
+        String[] splitAddress = address.split(", ");
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        this.block = new Block(splitAddress[0], isPrivate);
```

Comment:
Not sure does it make sense to add isPrivate to each object since we can know whether the address is private address by calling the Address class already.